numba/cuda/cudadrv/devicearray.py

```diff
@@ -680,8 +680,8 @@ def device_setup(self, gpu_data, stream=0):
 
 def from_array_like(ary, stream=0, gpu_data=None):
     "Create a DeviceNDArray object that is like ary."
-    if ary.ndim == 0:
-        ary = ary.reshape(1)
+    if isinstance(ary, np.generic):
+        ary = np.array(ary)
```

Comment:
These lines may not be needed at all, I'm not sure

Reply:
@gmarkall: is the input guaranteed to be an array here?

Would you be able to run a local test with these lines removed?

```suggestion
```

Reply:
I'll try a local test with them removed - I wonder if the input could sometimes be a record or `void`.

Reply:
That was my thought too.

Reply:
With those lines removed, it seems to introduce no new failures. I am seeing these errors:

```
======================================================================
ERROR: test_result_on_device (numba.cuda.tests.cudapy.test_reduction.TestReduction)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/nfs/gmarkall/numbadev/numba/numba/cuda/tests/cudapy/test_reduction.py", line 70, in test_result_on_device
    res = sum_reduce(A, res=got)
  File "/home/nfs/gmarkall/numbadev/numba/numba/cuda/kernels/reduction.py", line 251, in __call__
    not full_blockct)
  File "/home/nfs/gmarkall/numbadev/numba/numba/cuda/compiler.py", line 759, in __call__
    cfg(*args)
  File "/home/nfs/gmarkall/numbadev/numba/numba/cuda/compiler.py", line 523, in __call__
    sharedmem=self.sharedmem)
  File "/home/nfs/gmarkall/numbadev/numba/numba/cuda/compiler.py", line 597, in _kernel_call
    cu_func(*kernelargs)
  File "/home/nfs/gmarkall/numbadev/numba/numba/cuda/cudadrv/driver.py", line 1906, in __call__
    self.sharedmem, streamhandle, args)
  File "/home/nfs/gmarkall/numbadev/numba/numba/cuda/cudadrv/driver.py", line 1950, in launch_kernel
    None)
  File "/home/nfs/gmarkall/numbadev/numba/numba/cuda/cudadrv/driver.py", line 296, in safe_cuda_api_call
    self._check_error(fname, argstr, retcode)
  File "/home/nfs/gmarkall/numbadev/numba/numba/cuda/cudadrv/driver.py", line 331, in _check_error
    raise CudaAPIError(retcode, msg)
numba.cuda.cudadrv.driver.CudaAPIError: [701] Call to cuLaunchKernel(c_void_p(94560917514048), 1, 1, 1, 10, 1, 1, 0, None, <numba.cuda.cudadrv.driver.c_void_p_Array_16 object at 0x7f54fc0c4320>, None) results in CUDA_ERROR_LAUNCH_OUT_OF_RESOURCES

======================================================================
ERROR: test_sum_reduce (numba.cuda.tests.cudapy.test_reduction.TestReduction)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/nfs/gmarkall/numbadev/numba/numba/cuda/tests/cudapy/test_reduction.py", line 30, in test_sum_reduce
    self._sum_reduce(n)
  File "/home/nfs/gmarkall/numbadev/numba/numba/cuda/tests/cudapy/test_reduction.py", line 15, in _sum_reduce
    got = sum_reduce(A)
  File "/home/nfs/gmarkall/numbadev/numba/numba/cuda/kernels/reduction.py", line 251, in __call__
    not full_blockct)
  File "/home/nfs/gmarkall/numbadev/numba/numba/cuda/compiler.py", line 759, in __call__
    cfg(*args)
  File "/home/nfs/gmarkall/numbadev/numba/numba/cuda/compiler.py", line 523, in __call__
    sharedmem=self.sharedmem)
  File "/home/nfs/gmarkall/numbadev/numba/numba/cuda/compiler.py", line 597, in _kernel_call
    cu_func(*kernelargs)
  File "/home/nfs/gmarkall/numbadev/numba/numba/cuda/cudadrv/driver.py", line 1906, in __call__
    self.sharedmem, streamhandle, args)
  File "/home/nfs/gmarkall/numbadev/numba/numba/cuda/cudadrv/driver.py", line 1950, in launch_kernel
    None)
  File "/home/nfs/gmarkall/numbadev/numba/numba/cuda/cudadrv/driver.py", line 296, in safe_cuda_api_call
    self._check_error(fname, argstr, retcode)
  File "/home/nfs/gmarkall/numbadev/numba/numba/cuda/cudadrv/driver.py", line 331, in _check_error
    raise CudaAPIError(retcode, msg)
numba.cuda.cudadrv.driver.CudaAPIError: [701] Call to cuLaunchKernel(c_void_p(94560917514048), 1, 1, 1, 1, 1, 1, 0, None, <numba.cuda.cudadrv.driver.c_void_p_Array_16 object at 0x7f54fc3c5200>, None) results in CUDA_ERROR_LAUNCH_OUT_OF_RESOURCES

----------------------------------------------------------------------
```

but I'm pretty sure that's because of #4954 messing up tests for CUDA at the moment.

Reply:
Update: Removing these lines seems not to cause any issues.